numpy/core/src/multiarray/cblasfuncs.c

```diff
@@ -663,6 +673,17 @@ cblas_matrixproduct(int typenum, PyArrayObject *ap1, PyArrayObject *ap2,
                 goto fail;
             }
         }
+        if ((ap1shape == _row && PyArray_STRIDE(ap1, 1) == 0) ||
+            (ap1shape != _row && PyArray_STRIDE(ap1, 0) == 0)) {
```

Comment:
We do extra indentation here to make it visible that it is not part of the if body (I know it kills alignment here).